lib/connection.js

```diff
@@ -388,26 +390,26 @@ Connection.prototype._processHandshakeRequest = function(packet, keys) {
 
   this.application = application;
 
-  const username = keys[1];
-  const password = packet[username];
+  let authStrategy = keys[1];
+  const credentials = authStrategy && packet[authStrategy];
+  authStrategy = authStrategy || 'anonymous';
 
   this._emitPacketEvent('handshakeRequest', packet, packet.handshake[0], {
     packetType: 'handshake',
     handshakeRequest: true,
-    username,
-    password
+    authStrategy
   });
 
-  const callback = this._onSessionCreated.bind(this, username);
-  this.server.startSession(this, application, username, password, callback);
+  this.server.startSession(this, application, authStrategy, credentials,
+    this._onSessionCreated.bind(this));
 };
 
 // Callback of authentication operation
 //   username - user login, if any
```

Comment:
You can probably change order of arguments here as well, to correspond with function signature.